# 모델 설계 및 학습

사용할 예측 모델

1. Logistic Regression

2. Support Vector Machine (SVM)

3. K-Nearest Neighbor (KNN)

4. Random Forest

5. Naive Bayes

In [56]:
import pandas as pd

In [57]:
train_data = pd.read_csv('./input/train_data.csv')
train_label = pd.read_csv('./input/train_label.csv')

test_data = pd.read_csv('./input/test_data.csv')

In [58]:
train_data.shape, train_label.shape, test_data.shape

((891, 7), (891, 1), (418, 8))

- test data의 PassengerId가 필요해서 test_data에 id값을 넣어서 저장했고 이를 아래에서 분리 시켜 사용했다.

In [59]:
# test data의 PassengerId
test_data_PassengerId = test_data['PassengerId']

In [60]:
# test data에서 PassengerId를 제외한것
test_data = test_data.drop("PassengerId", axis=1)

In [62]:
# 제대로 분리가 됬는지 확인
test_data.shape, train_data.shape

((418, 7), (891, 7))

In [63]:
from sklearn.linear_model import LogisticRegression
from sklearn.svm import SVC
from sklearn.neighbors import KNeighborsClassifier
from sklearn.ensemble import RandomForestClassifier
from sklearn.naive_bayes import GaussianNB

from sklearn.utils import shuffle

In [64]:
# data shuffle
train_data, train_label = shuffle(train_data, train_label, random_state=5)

In [65]:
def train_and_test(model):
    model.fit(train_data, train_label)
    prediction = model.predict(test_data)
    accuracy = round(model.score(train_data, train_label)*100, 2)
    print("Accuracy: ", accuracy, "%")

    return prediction

In [66]:
# Logistic Regression
log_pred = train_and_test(LogisticRegression())

Accuracy:  78.45 %


In [67]:
# SVM
sum_pred = train_and_test(SVC())

Accuracy:  83.5 %


In [68]:
# KNN
knn_pred_4 = train_and_test(KNeighborsClassifier(n_neighbors=4))

Accuracy:  84.85 %


In [69]:
# Random Forest
rf_pred = train_and_test(RandomForestClassifier(n_estimators=100))

Accuracy:  88.55 %


In [70]:
# Naive Bayes
nb_pred = train_and_test(GaussianNB())

Accuracy:  78.45 %


Random Forest의 acc가 가장 높게 나왔다.

88.55%

# 마무리

- csv 파일로 만들어서 kaggle에 제출하기

In [74]:
rf_submission = pd.DataFrame({
    "PassengerId": test_data_PassengerId,
    "Survived": rf_pred
})

In [73]:
rf_submission.to_csv('./out/rf_submission.csv', index=False)